lib/rspec/rails/matchers/base_matcher.rb

```diff
@@ -0,0 +1,184 @@
+module RSpec
+  module Rails
+    module Matchers
+      # @ api private
+      class BaseMatcher
+        include RSpec::Matchers::Composable
+
+        # @api private
+        # Used to detect when no arg is passed to `initialize`.
+        # `nil` cannot be used because it's a valid value to pass.
+        UNDEFINED = Object.new.freeze
+
+        # @private
+        attr_reader :actual, :expected, :rescued_exception
+
+        # @private
+        attr_writer :matcher_name
+
+        def initialize(expected=UNDEFINED)
+          @expected = expected unless UNDEFINED.equal?(expected)
+        end
+
+        # @api private
+        # Indicates if the match is successful. Delegates to `match`, which
+        # should be defined on a subclass. Takes care of consistently
+        # initializing the `actual` attribute.
+        def matches?(actual)
+          @actual = actual
+          match(expected, actual)
+        end
+
+        # @api private
+        # Used to wrap a block of code that will indicate failure by
+        # raising one of the named exceptions.
+        #
+        # This is used by rspec-rails for some of its matchers that
+        # wrap rails' assertions.
+        def match_unless_raises(*exceptions)
+          exceptions.unshift Exception if exceptions.empty?
+          begin
+            yield
+            true
+          rescue *exceptions => @rescued_exception
+            false
+          end
+        end
+
+        # @api private
+        # Generates a description using {EnglishPhrasing}.
```

Comment:
Does this have to be fully qualified for yard to pick it up now? Thats whats failing the build?

Reply:
With: 
```diff
--- a/lib/rspec/rails/matchers/base_matcher.rb
+++ b/lib/rspec/rails/matchers/base_matcher.rb
@@ -46,7 +46,7 @@ module RSpec
         end

         # @api private
-        # Generates a description using {EnglishPhrasing}.
+        # Generates a description using {RSpec::Matchers::EnglishPhrasing}.
```

Output:
```sh
$   bin/yard doc --no-cache | ruby -e "                                                                                                   [17:40:29]
    while line = gets
      has_warnings ||= line.start_with?('[warn]:')
      has_errors   ||= line.start_with?('[error]:')
      puts line
    end
    if has_warnings || has_errors
      puts \"\n\nYARD emitted documentation warnings or errors.\"
      exit(1)
    end
  "
/Users/benoit.tigeot/.rvm/gems/ruby-2.5.3@appaloosa/gems/yard-0.8.7.6/lib/yard/parser/ruby/legacy/ruby_lex.rb:668: warning: constant ::FALSE is deprecated
/Users/benoit.tigeot/.rvm/gems/ruby-2.5.3@appaloosa/gems/yard-0.8.7.6/lib/yard/parser/ruby/legacy/ruby_lex.rb:620: warning: constant ::TRUE is deprecated
/Users/benoit.tigeot/.rvm/gems/ruby-2.5.3@appaloosa/gems/yard-0.8.7.6/lib/yard/parser/ruby/legacy/ruby_lex.rb:668: warning: constant ::FALSE is deprecated
Files:          31
Modules:        30 (    0 undocumented)
Classes:         9 (    0 undocumented)
Constants:       6 (    0 undocumented)
Methods:        85 (    0 undocumented)
 100.00% documented
```